sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/kafka/KafkaOAuth2AuthenticateCallbackHandler.java

```diff
@@ -0,0 +1,94 @@
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+
+/**
+ * {@link AuthenticateCallbackHandler} implementation for OAuth2 authentication with Azure Event Hubs.
+ */
+public class KafkaOAuth2AuthenticateCallbackHandler implements AuthenticateCallbackHandler {
+    private final AzureThirdPartyServiceProperties properties = new AzureThirdPartyServiceProperties();
+    private final DefaultAzureCredentialBuilderFactory defaultAzureCredentialBuilderFactory =
+        new DefaultAzureCredentialBuilderFactory(properties);
+    private TokenCredential credential;
+    private AzureOAuthBearerToken accessToken;
+    private String tokenAudience;
+
+    private final AzureTokenCredentialResolver tokenCredentialResolver = new AzureTokenCredentialResolver();
+
+    @Override
+    public void configure(Map<String, ?> configs, String mechanism, List<AppConfigurationEntry> jaasConfigEntries) {
+        String bootstrapServer = Arrays.asList(configs.get(BOOTSTRAP_SERVERS_CONFIG)).get(0).toString();
+        bootstrapServer = bootstrapServer.replaceAll("\\[|\\]", "");
+        URI uri = URI.create("https://" + bootstrapServer);
+        this.tokenAudience = uri.getScheme() + "://" + uri.getHost();
+        credential = (TokenCredential) configs.get(AZURE_TOKEN_CREDENTIAL);
+        AzureIdentityCustomConfigUtils.convertConfigMapToAzureProperties(configs, properties);
+    }
+
+    @Override
+    public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
+        for (Callback callback : callbacks) {
+            if (callback instanceof OAuthBearerTokenCallback) {
+                OAuthBearerTokenCallback oauthCallback = (OAuthBearerTokenCallback) callback;
+                credential = getTokenCredential();
+                OAuthBearerToken token = getOAuthBearerToken();
+                oauthCallback.token(token);
+            } else {
+                throw new UnsupportedCallbackException(callback);
+            }
+        }
+    }
+
+    private TokenCredential getTokenCredential() {
+        if (credential == null) {
+            // Resolve the token credential when there is no credential passed from configs.
+            credential = tokenCredentialResolver.resolve(properties);
+            if (credential == null) {
+                // Create DefaultAzureCredential when no credential can be resolved from configs.
+                credential = defaultAzureCredentialBuilderFactory.build().build();
+            }
+        }
+        return credential;
+    }
+
+    private OAuthBearerToken getOAuthBearerToken() {
+        if (accessToken == null || accessToken.isExpired()) {
+            TokenRequestContext request = new TokenRequestContext();
+            request.addScopes(tokenAudience);
+            request.setTenantId(properties.getProfile().getTenantId());
+            AccessToken accessToken = credential.getToken(request).block(Duration.ofSeconds(30));
```

Comment:
how about use a const for : Duration.ofSeconds(30) ?

Reply:
I think this is better

Reply:
How about this?